utils/string/README.md

```diff
@@ -0,0 +1,33 @@
+`icu_string` is a utility crate of the [`ICU4X`] project.
+
+It provides a collection of helper utilities for handling Rust strings.
+
+## Example
+
+```rust
+use icu_string::Slice;
+
+/// Parse an input string slice and return a name parsed out of it.
+///
+/// # Type parameters
+///
+/// - `S`: A type of a slice provided as an input: `&str`, `String` or `Cow<str>`.
+/// - `RS`: Return slice: `&str`, `String` or `Cow<str>`.
```

Comment:
Given that the ICU4X style guide includes `SmallString` and `TinyStr` as first-class string types, it seems like we should include them here.